File: Problem/Programmers/DFS_Q2.py
answer = 0

def DFS(idx, computers, isVisited):
    global answer

    if isVisited[idx]:
        return

    isVisited[idx] = True

    for i in range(len(computers[idx])):
        if not isVisited[i] and computers[idx][i] == 1:
            DFS(i, computers, isVisited)


def solution(n, computers):
    global answer
    isVisited = [0] * n

    for idx in range(n):
        if not isVisited[idx]:
            answer += 1
            DFS(idx, computers, isVisited)
    return answer

File: Problem/Programmers/dfs_bfs/Dfs_Q2.java

// 네트워크
class Solution {
    boolean[] isVisited;
    
    public void dfs(int n, int[][] computers, int curIdx){
        if(isVisited[curIdx]){
            return;
        }
        isVisited[curIdx] = true;
        
        for(int i=0;i<computers[curIdx].length;i++){
            if(computers[curIdx][i] == 1 && !isVisited[i]){
                dfs(n, computers, i);
            }
        }
    }
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        isVisited = new boolean[n];
        
        for(int i=0;i<computers.length;i++){
            if(!isVisited[i]){
                dfs(n, computers, i);
                answer++;
            }
        }
        
        return answer;
    }
}